client/components/UI/DateTime.tsx

```diff
@@ -12,37 +12,82 @@ import './style.scss';
  * @name DateTime
  * @description DateTime component to display text formatted date and time
  */
-function DateTime({date, withTime, withDate, withYear, padLeft, toBeConfirmed}) {
+function DateTime({
+    date,
+    withTime,
+    withDate,
+    withYear,
+    padLeft,
+    toBeConfirmed,
+    allDay,
+    isEndEventDateTime,
+    noEndTime,
+    setHideDash,
+    multiDay,
+}) {
     const {gettext} = superdeskApi.localization;
     const dateFormat = appConfig.planning.dateformat;
     const timeFormat = appConfig.planning.timeformat;
-    const newTimeFormat = toBeConfirmed ? `[${gettext(`Time ${gettext('TBC')}`)}]` : timeFormat;
-    let format = withYear ? dateFormat : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
+    const newTimeFormat = toBeConfirmed
+        ? `[${gettext(`Time ${gettext('TBC')}`)}]`
+        : timeFormat;
+    let format = withYear
+        ? dateFormat
+        : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
     let dateTimeFormat = [
         withDate ? format : null,
         withTime ? newTimeFormat : null,
-    ].filter((d) => d).join('\u00a0'); // &nbsp;
+    ]
+        .filter((d) => d)
+        .join('\u00a0'); // &nbsp;
+
+    let eventEndDate;
+
+    if (noEndTime && !multiDay) {
+        eventEndDate = null;
+        setHideDash && setHideDash(true);
+    } else if (allDay && multiDay) {
+        eventEndDate = moment(date).format(dateTimeFormat)
+            .slice(0, 5) || '';
+    } else if (noEndTime && multiDay) {
+        eventEndDate = moment(date).format(dateTimeFormat)
+            .slice(0, 5) || '';
+    } else if (allDay && !multiDay) {
+        setHideDash && setHideDash(true);
+    } else {
+        eventEndDate = moment(date).format(dateTimeFormat);
+    }
+
+    let eventStartDate;
+
+    if (allDay && multiDay) {
+        eventStartDate = moment(date).format(dateTimeFormat)
```

Comment:
when ignoring the time there should be always `moment.utc` otherwise this will be converted to local timezone